app/main.py

```
@@ -1,3 +1,68 @@
+from typing import Any, Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", float, int]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Union["Distance", float, int]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        if isinstance(other, (int, float)):
+            self.km += other
+            return self
+
+    def __mul__(self, other: Union[(int, float)]) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                km=self.km * other
+            )
+
+    def __truediv__(self, other: Union[int, float]) -> "Distance":
+        if isinstance(other, (int, float)):
+            if other != 0:
+                new_km = self.km / other
+                return Distance(round(new_km, 2))
```

`__truediv__` currently returns a `Distance` only when `other != 0`; when `other == 0` the method falls through and returns `None`. The description requires that division produce a `Distance` instance (rounded to 2 decimals). Quote from the description: "The resulting km must be rounded to 2 decimal places ... Result must be an instance of Distance." Handle zero explicitly (raise `ZeroDivisionError` or let the division raise) so the method never silently returns `None` for invalid inputs.